hs_collection_resource/tests/test_collection.py

```diff
@@ -0,0 +1,414 @@
+
+        # make resGen1 public
+        url_to_set_resource_flag_for_resGen1 = self.url_to_set_resource_flag.format(self.resGen1.short_id)
+        response = self.api_client.post(url_to_set_resource_flag_for_resGen1, {'t': 'make_public'}, HTTP_REFERER='http://foo/bar')
+
+        # check resGen1 new  sharing status --> public
+        self.assertEqual(BaseResource.objects.get(short_id=self.resGen1.short_id).raccess.public, True)
+        self.assertEqual(BaseResource.objects.get(short_id=self.resGen1.short_id).raccess.discoverable, True)
+        ## the following two statements return wrong sharing status. DONT KNOW WHY!!!
+        # self.assertEqual(self.resGen1.raccess.public, True)
+        # self.assertEqual(self.resGen1.raccess.discoverable, True)
+
+        # check resCollection current sharing status --> private
+        self.assertEqual(BaseResource.objects.get(short_id=self.resCollection.short_id).raccess.public, False)
+        self.assertEqual(BaseResource.objects.get(short_id=self.resCollection.short_id).raccess.discoverable, False)
+        self.assertEqual(self.resCollection.raccess.public, False)
+        self.assertEqual(self.resCollection.raccess.discoverable, False)
+
+        # check resCollection can_be_public_or_discoverable again --> False
+        # collection does not need res files
+        self.assertTrue(self.resCollection.has_required_content_files())
+        # collection should have at least on member resource
+        self.assertFalse(self.resCollection.metadata.has_all_required_elements())
+        # False
+        self.assertFalse(self.resCollection.can_be_public_or_discoverable)
+
+        # add one public resGen1 into resCollection
+        url_to_update_resCollection = self.url_to_update_collection.format(self.resCollection.short_id)
+        response = self.api_client.post(url_to_update_resCollection,
+                                        {'resource_id_list': [self.resGen1.short_id]},
+                                        HTTP_REFERER='http://foo/bar')
+        resp_json = json.loads(response.content)
+        self.assertEqual(resp_json["status"], "success")
+        self.assertEqual(resp_json["metadata_status"], "Sufficient to make public")
+        self.assertEqual(self.resCollection.metadata.collection.resources.all().count(), 1)
+
+        # make resCollection public
+        url_to_set_resource_flag_for_resCollection = self.url_to_set_resource_flag.format(self.resCollection.short_id)
+        response = self.api_client.post(url_to_set_resource_flag_for_resCollection, {'t': 'make_public'}, HTTP_REFERER='http://foo/bar')
+
+        # check resCollection current sharing status --> pubilic
+        self.assertEqual(BaseResource.objects.get(short_id=self.resCollection.short_id).raccess.public, True)
+        self.assertEqual(BaseResource.objects.get(short_id=self.resCollection.short_id).raccess.discoverable, True)
+        ## again, the following two statements return wrong answers
```

Comment:
Since you have provided these comments above, you can remove these repeated comments in other places.